### DISCUSSION TITLE
AppleTV 4k stops being responsive

### DISCUSSION THREAD

**Discussion by Deanerh** [PRProd/HA-Firemote#380] [General]
Have a problem where when I haven’t used the remote card in a while it becomes unresponsive. What’s weird is sometimes it’s all of the buttons while occasionally the control buttons won’t work, but the shortcut buttons do. Any thoughts? Only thing that seems to work is restating the server. Just reloading the configuration doesn’t.

**Comment by PRProd** [comment]
Greetings @Deanerh!

I've noticed something similar happening with my Amazon Fire TV devices as well.  I suspect that our issues might be related to a problem with some Home Assistant entities not maintaining a consistent connection with our devices.  

Here's how I get there:

It's useful to understand what Firemote is, and what it isn't.  I wrote a section on the [Firemote wiki](https://github.com/PRProd/HA-Firemote/wiki) homepage called [How Does Firemote Send commands to my devices?](https://github.com/PRProd/HA-Firemote/wiki#how-does-firemote-send-commands-to-my-devices).  In that section you will learn that Firemote is not an integration, it's just a card.  That means it relies on existing integration(s) to control your Apple TV (or other) device.

When you configured your Firemote card to control your apple TV device, you might remember that you had to attach two different entities:

![image](https://github.com/PRProd/HA-Firemote/assets/25486599/2b592fef-0662-4a69-aaf5-b64e96cc0c59)

Depending on the button you click on your Firemote, the card will send a command to *one or the other* of these entities.  If one of the entities/integrations isn't behaving properly, some buttons will work on your Firemote, while others will not.  

Based on your description, it sounds like the **Apple TV Media Player Entity** remains connected, allowing you to switch sources with your shortcut buttons, but the **Associated Apple TV Remote Entity** seems to lose its connection with your Apple TV device.

Why is this happening?  I'm not sure, but I do have an idea for you to try that will be faster than restarting your Home Assistant server.

Try this ~ Click on **Settings** -> **Devices & Services**
![image](https://github.com/PRProd/HA-Firemote/assets/25486599/8aad4da6-6df3-4a4d-8a89-caf6f7f4b909)

**Integrations** -> **Apple TV**
![image](https://github.com/PRProd/HA-Firemote/assets/25486599/e706fe06-d465-4eeb-afdc-b0238ca3bb86)

**Click the 3 dots** -> **Reload**
![image](https://github.com/PRProd/HA-Firemote/assets/25486599/1f83f020-6b00-4852-95c8-7d19e00f545e)
<br>

I'm interested to know if that helps with your issues or not.  Please let me know.  Thanks!

-Doug

**Comment by Deanerh** [comment]
That worked!! Bought you a few cups of coffee. Great card and I really appreciate your support
________________________________

**Reply by PRProd** [reply]
THANK YOU!  I'm so glad it worked for you!

**Comment by Deanerh** [comment]
Is there anyway in HA to have it reload the AppleTV integration periodically in the background?
________________________________

**Reply by PRProd** [reply]
That's a really great question!  I haven't done any research yet to see if it's possible to do it through a script or anything, but I'll look soon and let you know.

**Comment by Deanerh** [comment]
So first, let me clarify, I don’t really know YAML and when I do actually go beyond the visually editor, it’s pretty much monkey-see-monkey-do by searching for posts where someone has a solved a similar issue.  I did that last night using the Automation idea you suggested.    Here’s the code for the Automation.  So far it’s stayed connected.  I’ll see how it does after 24 hours.

alias: Reload AppleTV integration every 1 minute
description: ""
trigger:
  - platform: time_pattern
    minutes: "1"
condition: []
action:
  - service: homeassistant.reload_config_entry
    data: {}
    target:
      entity_id:
        - media_player.asheville_den
        - remote.asheville_den
mode: single

________________________________

**Reply by PRProd** [reply]
Awesome!  Thanks for sharing that!  Please let me know if it keeps working well over the next few days.  I might try it too!  

I think I might also need to see if an issue has been opened/reported to the HA team for this already.  If not, I will open one.

**Comment by Deanerh** [comment]
Sigh, it’s not 100%.  This morning when I tried it, all was good.  Just tried again and the Power button worked (my TV, soundbar and ATV are controlled via ARC) to turn all devices on, but then the control pad and volume buttons didn’t.  A reload got it all working again.
________________________________

**Reply by PRProd** [reply]
Bummer!  Well, it was a band-aid anyway, right?  You probably just happened to catch it at a bad time, between a disconnect and before the next 1 minute "reload" interval.

**Reply by PRProd** [reply]
Out of curiosity, do you have more Apple TV entities configured other than the  **media_player.asheville_den** and the **remote.asheville_den**?  I started to look at my setup and was wondering if some of my issues were because I have so many entities, and many of them are not currently online.  Is your situation similar?

**Reply by PRProd** [reply]
This open issue is close to what we are experiencing.  I didn't read the entire thread yet because it already has 76 comments...
https://github.com/home-assistant/core/issues/69832

**Comment by Deanerh** [comment]
I read through and it seemed the main issue was  that the TCL TV was the culprit in terms of it was shutting off the connection to HA after some inactivity. One thing that seemed to help was there was a setting on the TCL that seemed to keep the connection open at last somewhat better.

I only have 2 entities for the ATV: media.control and remote. Do you have more?
________________________________

**Reply by PRProd** [reply]
I only have 2 for Apple TV as well, but I haven't been experiencing connectivity issues with Apple TV.  I've noticed this issue with my Amazon Fire TV connected through the ADB integration.

Since I've made it a habit to actually own the devices that I support, I have MANY entities in mine.  Most of the entities I have are not always online, since they spend a lot of time in a drawer.  When I need to test something, I'll connect it for a while, and then put it away again, which is why I still keep the disconnected entities in my config.  That's why mine looks like this:
![image](https://github.com/PRProd/HA-Firemote/assets/25486599/bccec4e6-ab64-402b-b0ec-2f19ef69238f)

**Comment by Deanerh** [comment]
Trying something different with that automation.  Like I mentioned, I’m not good with YAML, so I used the Visual Editor to edit the automation once I pasted the code from the thread I found.  I had put “1” in the minutes option in the VE interface, assuming that would trigger it every minute.
I just went and took a look at the automation and it said something like “trigger on the 1st minute of every hour”, which is not what I was wanting.  I deleted the “1” and put “59” in the seconds field (it would not let me do “60”) and now it says “Trigger on the 59th second of every minute of every hour”, which is what I want.
We’ll see how this does.
________________________________
